fastly/base_fastly_service_v1.go

```diff
@@ -147,6 +148,25 @@ func resourceService(serviceDef ServiceDefinition) *schema.Resource {
 		UpdateContext: resourceUpdate(serviceDef),
 		DeleteContext: resourceDelete(serviceDef),
 		Importer:      resourceImport(serviceDef),
+		CustomizeDiff: customdiff.All(
+			customdiff.ComputedIf("cloned_version", func(_ context.Context, d *schema.ResourceDiff, _ interface{}) bool {
+				// If anything other than name, comment and version_comment has changed, the current version will be
+				// cloned in resourceServiceUpdate so set it as recomputed. These three fields can be updated without
+				// creating a new version
+				for _, changedKey := range d.GetChangedKeysPrefix("") {
+					if changedKey == "name" || changedKey == "comment" || changedKey == "version_comment" {
+						continue
+					}
+					return true
+				}
+				return false
+			}),
+			customdiff.ComputedIf("active_version", func(_ context.Context, d *schema.ResourceDiff, _ interface{}) bool {
+				// If cloned_version is recomputed and we are automatically activating new versions (controlled with the
+				// activate flag) then the active_version will be recomputed too.
+				return d.HasChange("cloned_version") && d.Get("activate") == true
```

Comment:
Just had a thought here - might need to check `d.HasChange("activate")`, as this could change the active version too. (Could it? Or does there need to be another change for this to take effect?)

Reply:
Nope, the service is only activated if there's a change in a nested block. If the activate block was outside of this condition then the following could be applicable but it isn't.

```diff
diff --git a/fastly/base_fastly_service_v1.go b/fastly/base_fastly_service_v1.go
index 3ca10b3f..9982cdaa 100644
--- a/fastly/base_fastly_service_v1.go
+++ b/fastly/base_fastly_service_v1.go
@@ -164,7 +164,10 @@ func resourceService(serviceDef ServiceDefinition) *schema.Resource {
 			customdiff.ComputedIf("active_version", func(_ context.Context, d *schema.ResourceDiff, _ interface{}) bool {
 				// If cloned_version is recomputed and we are automatically activating new versions (controlled with the
 				// activate flag) then the active_version will be recomputed too.
-				return d.HasChange("cloned_version") && d.Get("activate") == true
+				updatingAndActivating := d.HasChange("cloned_version") && d.Get("activate") == true
+				activatingExisting := d.Get("active_version") != d.Get("cloned_version") && d.HasChange("activate")
+
+				return updatingAndActivating || activatingExisting
 			}),
 		),
 ```